airbyte-connector-builder/connector_builder/generated/apis/default_api_interface.py

```diff
@@ -0,0 +1,125 @@
+# This file was auto-generated from Airbyte's custom OpenAPI templates
+# coding: utf-8
+
+import inspect
+from abc import ABC, abstractmethod
+from typing import Callable, Dict, List  # noqa: F401
+
+from fastapi import (  # noqa: F401
+    APIRouter,
+    Body,
+    Cookie,
+    Depends,
+    Form,
+    Header,
+    Path,
+    Query,
+    Response,
+    Security,
+    status,
+)
+
+from connector_builder.generated.models.extra_models import TokenModel  # noqa: F401
+
+
+from connector_builder.generated.models.invalid_input_exception_info import InvalidInputExceptionInfo
+from connector_builder.generated.models.known_exception_info import KnownExceptionInfo
+from connector_builder.generated.models.stream_read import StreamRead
+from connector_builder.generated.models.stream_read_request_body import StreamReadRequestBody
+from connector_builder.generated.models.streams_list_read import StreamsListRead
+from connector_builder.generated.models.streams_list_request_body import StreamsListRequestBody
+
+
+class DefaultApi(ABC):
+    @abstractmethod
+    async def get_manifest_template(
+        self, 
+    ) -> str:
+        """Return a connector manifest template to use as the default value for the yaml editor"""
+
+    @abstractmethod
+    async def list_streams(
+        self, 
+        streams_list_request_body: StreamsListRequestBody = Body(None, description=""),
+    ) -> StreamsListRead:
+        """List all streams present in the connector manifest, along with their specific request URLs"""
+
+    @abstractmethod
+    async def read_stream(
+        self, 
+        stream_read_request_body: StreamReadRequestBody = Body(None, description=""),
+    ) -> StreamRead:
+        """Reads a specific stream in the source. TODO in a later phase - only read a single slice of data."""
+
+
+def _assert_signature_is_set(method: Callable) -> None:
+    """
+    APIRouter().add_api_route expects the input method to have a signature. It gets signatures
+    by running inspect.signature(method) under the hood. 
+
+    In the case that an instance method does not declare "self" as an input parameter (due to developer error 
+    for example), then the call to inspect.signature() fails. 
+
+    This method is sort of a workaround. It tells the developer how to fix the problem if it's detected.
+    """
+    try:
+        inspect.signature(method)
+    except ValueError as e:
+        # Based on empirical observation, the call to inspect fails with a ValueError
+        # with exactly one argument: "invalid method signature"
+        if e.args and len(e.args) == 1 and e.args[0] == "invalid method signature":
+            # I couldn't figure out how to setattr on a "method" object to populate the signature. For now just kick
```

Comment:
What do you mean by setattr? Do you mean [setting the arguments passed to a function object](https://docs.python.org/3/library/functools.html#functools.partial)?

Reply:
the "automagic" way of fixing the issue described in the method docstring, would have been to do `setattr(method, "__signature__", <something>)` but that's not possible because instance methods are `object` subclasses, and [you can't use setattr on `object`s](https://stackoverflow.com/a/12839070/3237889)

So I couldn't figure out how to setattr on the method in the 1 day timeblock I allocated for this. 

The workaround is to raise an exception at runtime. This is good enough because the error will be detected immediately when the developer tries to run the server, so builds should very quickly fail

Reply:
I added this in a comment as well